flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/StreamTaskStateInitializerImpl.java

```diff
@@ -227,6 +238,24 @@ public StreamOperatorStateContext streamOperatorStateContext(
                 timeServiceManager = null;
             }
 
+            // Add stats for input channel and result partition state
+            Stream.concat(
+                            prioritizedOperatorSubtaskStates.getPrioritizedInputChannelState()
+                                    .stream(),
+                            prioritizedOperatorSubtaskStates.getPrioritizedResultSubpartitionState()
+                                    .stream())
+                    .filter(Objects::nonNull)
+                    .forEach(channelHandle -> channelHandle.collectSizeStats(statsCollector));
+
+            // Report collected stats to metrics
+            statsCollector
+                    .getStats()
+                    .forEach(
```

Comment:
If you are going all in with streams, why don't you merge those two?

Reply:
One is still collecting remaining stats from channels, the second is reporting. I like to keep them separate.